crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -588,59 +554,40 @@ where
         let deleted_objects = data
             .transactions
             .iter()
-            .flat_map(|tx| get_deleted_objects(&tx.effects))
+            .flat_map(|tx| tx.removed_object_refs_post_version())
             .collect::<Vec<_>>();
         let indexed_deleted_objects: Vec<IndexedDeletedObject> = deleted_objects
             .into_iter()
-            .map(|o| IndexedDeletedObject {
-                object_id: o.0,
-                object_version: o.1.value(),
+            .map(|obj_ref| IndexedDeletedObject {
+                object_id: obj_ref.0,
+                object_version: obj_ref.1.into(),
                 checkpoint_sequence_number: checkpoint_seq,
             })
             .collect();
 
-        let (latest_objects, _) = get_latest_objects(data.output_objects());
-        let history_object_map = data
-            .output_objects()
+        let latest_live_output_objects = data.latest_live_output_objects();
+        let latest_live_output_object_map = latest_live_output_objects
+            .clone()
             .into_iter()
-            .map(|o| ((o.id(), o.version()), o.clone()))
+            .map(|o| (o.id(), o.clone()))
             .collect::<HashMap<_, _>>();
 
-        let history_objects: Vec<Object> = data
+        let output_objects = data
             .transactions
             .iter()
-            .flat_map(|tx| {
-                let CheckpointTransaction {
-                    transaction: tx,
-                    effects: fx,
-                    ..
-                } = tx;
-                fx.all_changed_objects()
-                    .into_iter()
-                    .map(|(oref, _owner, _kind)| {
-                        let history_object = history_object_map.get(&(oref.0, oref.1)).unwrap_or_else(|| {
-                            panic!(
-                                "object {:?} version {:?} not found in CheckpointData (tx_digest: {})",
-                                oref.0,
-                                oref.1,
-                                tx.digest()
-                            )
-                        });
-                        assert_eq!(oref.2, history_object.digest());
-                        history_object.clone()
-                    })
-                    .collect::<Vec<_>>()
-            })
-            .collect();
-
+            .flat_map(|tx| &tx.output_objects)
+            .collect::<Vec<_>>();
         let move_struct_layout_map =
-            get_move_struct_layout_map(&history_objects, package_resolver).await?;
```

Comment:
Please also add a TODO here to remove this later, since this is where the bug is actually is.